src/s3fs.cpp

```diff
@@ -393,78 +402,24 @@ static int get_object_attribute(const char* path, struct stat* pstbuf, headers_t
-    strpath     = path;
-    result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-    s3fscurl.DestroyCurlHandle();
-
-    // if not found target path object, do over checking
-    if(-EPERM == result){
-        // [NOTE]
-        // In case of a permission error, it exists in directory
-        // file list but inaccessible. So there is a problem that
-        // it will send a HEAD request every time, because it is
-        // not registered in the Stats cache.
-        // Therefore, even if the file has a permission error, it
-        // should be registered in the Stats cache. However, if
-        // the response without modifiying is registered in the
-        // cache, the file permission will be 0644(umask dependent)
-        // because the meta header does not exist.
-        // Thus, set the mode of 0000 here in the meta header so
-        // that s3fs can print a permission error when the file
-        // is actually accessed.
-        // It is better not to set meta header other than mode,
-        // so do not do it.
-        //
-        (*pheader)["x-amz-meta-mode"] = str(0);
-
-    }else if(0 != result){
-        if(overcheck){
-            // when support_compat_dir is disabled, strpath maybe have "_$folder$".
-            if('/' != strpath[strpath.length() - 1] && std::string::npos == strpath.find("_$folder$", 0)){
-                // now path is "object", do check "object/" for over checking
-                strpath    += "/";
-                result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-                s3fscurl.DestroyCurlHandle();
-            }
-            if(support_compat_dir && 0 != result){
-                // now path is "object/", do check "object_$folder$" for over checking
-                strpath.erase(strpath.length() - 1);
-                strpath    += "_$folder$";
-                result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-                s3fscurl.DestroyCurlHandle();
-
-              if(0 != result){
-                  // cut "_$folder$" for over checking "no dir object" after here
-                  if(std::string::npos != (Pos = strpath.find("_$folder$", 0))){
-                      strpath.erase(Pos);
-                  }
-              }
-            }
-        }
-        if(support_compat_dir && 0 != result && std::string::npos == strpath.find("_$folder$", 0)){
-            // now path is "object" or "object/", do check "no dir object" which is not object but has only children.
-            if('/' == strpath[strpath.length() - 1]){
-                strpath.erase(strpath.length() - 1);
-            }
-            if(-ENOTEMPTY == directory_empty(strpath.c_str())){
-                // found "no dir object".
-                strpath  += "/";
-                *pisforce = true;
-                result    = 0;
-            }
-        }
+    s3obj_list_t headlist;
+    std::string parent;
+    std::string child;
+    size_t pos = strpath.rfind('/');
```

Comment:
If strpath is `path /`, is there any effect in the subsequent processing?
(Will it be `path //` or `path / _ $ folder $`?)